File: CPP Data/T 01 Patterns/Pattern_06_Inverted_Numbered_Right_Pyramid.cpp
#include<bits/stdc++.h>
using namespace std;

// 1 2 3 4 5 6
// 1 2 3 4 5
// 1 2 3 4
// 1 2 3
// 1 2
// 1

void invertedNumberRightPyramidPattern(int n)
{
    for(int i = 1; i <= n; i++) {

        for(int j = 1; j <= n - i + 1; j++){

            cout << j << " ";
        }
        cout << endl;
    }
}

int main(){

    int n;
    cout << "Enter the number of rows for the inverted number right pyramid pattern: ";
    cin >> n;

    invertedNumberRightPyramidPattern(n);
    
    return 0;
}

File: 00-Patterns-Python/Pattern_06_Inverted_Numbered_Right_Pyramid.py
# 1 2 3 4 5 6
# 1 2 3 4 5
# 1 2 3 4
# 1 2 3
# 1 2
# 1

def inverted_number_right_pyramid_pattern(n):

    # Outer loop controls the rows. It runs 'n' times (once for each row)
    for i in range(1, n +1, 1):

        # Inner loop controls the columns (numbers printed in the current row
        # 'n - i + 1 times'
        for j in range(1, n - i + 2, 1):
            print(j, end=" ")

        # Move to the next line after each row is complete
        print()

num = input("Enter the number of rows for the inverted number right pyramid pattern: ")
inverted_number_right_pyramid_pattern(int(num))
